includes/html/graphs/generic_data.inc.php

```diff
@@ -18,7 +18,7 @@
 
 require 'includes/html/graphs/common.inc.php';
 
-$stacked = generate_stacked_graphs();
+$stacked = generate_stacked_graphs(! empty($port['ifSpeed']) && ($vars['port_speed_zoom'] ?? Config::get('graphs.port_speed_zoom')) == 1);
```

Comment:
I don't think the `== 1`  actually does anything in these.  The only case where it does is if someone sets port_speed_zoom=0 on the url, in that case I don't think we should respect it.

It might be more obvious to cast to a bool

Reply:
> '1' == true
= true

> '0' == true
= false

> '1' == 1
= true

> '0' == 1
= false

Reply:
When I added it, the toggle started working correctly. 🤷

I'll try replacing it with true.

Reply:
Give me an example graph and I'll test it.

Reply:
Go to a random port in an instance. Click the zoom to speed link on, then click it back off. As long as the Zoom to speed toggle is in the URL, even when off (0), it will still be zoomed to speed. When the "==1" is added it only does zoom to speed when URL toggle is 1.

Reply:
I dumped the value on a graph and when it is port_speed_zoom=0, for some reason the value in $vars['port_speed_zoom'] is "yes".

Reply:
Fixed, now none of the == 1 are needed
#14967 14967

Reply:
Updated my PR. I've applied your url handling fix, and my zoom to speed fix, and it's working as expected!